pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -279,40 +235,11 @@ func (c *Cluster) reconcileServices(activeDaemon string) error {
 
 	// enable monitoring if `monitoring: enabled: true`
 	if c.spec.Monitoring.Enabled {
-		if err := c.EnableServiceMonitor(activeDaemon); err != nil {
+		if err := c.EnableServiceMonitor(); err != nil {
 			return errors.Wrap(err, "failed to enable service monitor")
 		}
 	}
 
-	return c.updateServiceSelectors(activeDaemon)
-}
-
-// Make a best effort to update the services that have been labeled for being updated
-// when the mgr has changed. They might be services for node ports, ingress, etc
-func (c *Cluster) updateServiceSelectors(activeDaemon string) error {
```

Comment:
For the upgrade scenario, we need to keep this method, but change its implementation. Basically, if we find a selector for the DaemonIDLabel (line 303), we need to remove that selector label and update the service. But if that selector label is not found, we can skip the service and no update would be needed.